### PR TITLE
save click on publication

### PR DESCRIPTION
**Pull request by SimonLab**
ref: #35
This PR saved each click on a publication into postgres

### CODE REVIEW

File: web/templates/search/results.html.eex [hunk 1]
```diff
@@ -9,7 +9,7 @@
   <h2 class="f6 pb2 bw1 bb bep-b--blue"> <%= @data["total"] %> Results</h2>
   <ul class="ph1 list">
     <%= for {evidence, i} <- Enum.with_index(@data["documents"]) do %>
-      <li class="bl bw2 mv3 pl2 fw9 pr4"><%= i + 1 %>. <%= link evidence["title"], to: evidence["link"], class: "link bep-blue" %> </li>
+      <li class="bl bw2 mv3 pl2 fw9 pr4"><%= i + 1 %>. <%= link evidence["title"], to: evidence["link"], class: "link bep-blue publication", target: "_blank", searchId: @id %> </li>
```
Comment (Danwhy) [inline web/templates/search/results.html.eex]: I think this searchId should be a [data attribute](https://developer.mozilla.org/en-US/docs/Learn/HTML/Howto/Use_data_attributes)

File: web/static/js/publication.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
+module.exports = (function() {
+  var publications = document.querySelectorAll(".publication");
+
+  publications.forEach(function(publication) {
```
Comment (shouston3) [inline web/static/js/publication.js]: `forEach` on a nodelist [has some browser support issues](https://developer.mozilla.org/en-US/docs/Web/API/NodeList/forEach)

File: web/static/js/publication.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+        value: e.target.textContent
+      };
+
+      var request = new XMLHttpRequest();
+      request.open("POST", "/publication", true);
+      request.setRequestHeader("Content-type", "application/json");
+
+      request.onreadystatechange = function() {
+        if (request.readyState === 4 && request.status === 200) {
+            console.log(request.responseText);
```
Comment (shouston3) [inline web/static/js/publication.js]: Having:
```javascript
if (request.readyState === 4) {
  if (request.status === 200) {
    console.log(request.responseText);
  } else {
    console.log('STATUS: ', request.status, 'REQUEST: ', request)
  }
```

and passing back the error from the `PublicationController` `create` action would make debugging easier (if this errors in production)

File: web/static/js/publication.js (hunk 2 above)
Comment (SimonLab) [inline web/static/js/publication.js]: I wasn't sure if Phoenix was being able to compile to a polyfill version, but I think you're right and will use a simple for loop

**Comment by SimonLab** [comment]
@shouston3 I've just updated this PR based on your comments, this is now ready for review :+1:

**Pull request merged**